zhmcclient/_partition_link.py

```diff
@@ -0,0 +1,613 @@
+        return part_list
+
+    @logged_api_call
+    def delete(self, force_detach=False):
+        """
+        Delete this partition link on the HMC.
+
+        If there are active partitions to which the partition link is attached,
+        the operation will fail by default. The 'force_detach' parameter can be
+        used to forcefully detach the partition link from active partitions
+        before deleting it.
+
+        This method performs the "Delete Partition Link" HMC operation and
+        waits for completion of its asynchronous job.
+
+        HMC/SE version requirements:
+
+        * SE version >= 2.16.0
+        * for technology-specific support, see the API features described in
+          :ref:`Partition Links`
+
+        Authorization requirements:
+
+        * Object-access permission to this partition link.
+        * Object-access permission to all partitions that currently have this
+          partition link attached.
+        * Task permission to the "Delete Partition Link" task.
+
+        Parameters:
+
+          force_detach (bool): Controls what to do with active partitions
+            associated with this partition link. If True, such partitions
+            are detached forcefully. If False, the operation fails with
+            status code 409 (Conflict) and reason code 100.
+
+        Raises:
+
+          :exc:`~zhmcclient.HTTPError`
+          :exc:`~zhmcclient.ParseError`
+          :exc:`~zhmcclient.AuthError`
+          :exc:`~zhmcclient.ConnectionError`
+          :exc:`~zhmcclient.OperationTimeout`: The timeout expired while
+            waiting for completion of the operation.
+        """
+        body = {}
+        if force_detach:
+            body['force-detach'] = True
+
+        self.manager.session.post(
+            uri=f'{self.uri}/operations/delete',
+            resource=self,
+            body=body,
+            wait_for_completion=True)
+
+        # pylint: disable=protected-access
+        self.manager._name_uri_cache.delete(
+            self.get_properties_local(self.manager._name_prop, None))
+        self.cease_existence_local()
+
+    @logged_api_call
+    def update_properties(self, properties):
+        """
+        Update writeable properties of this partition link.
+
+        This method can be used to attach and detach this partition link
+        to and from partitions, by modifying the appropriate resource
+        properties.
+
+        This method serializes with other methods that access or change
+        properties on the same Python object.
+
+        This method performs the "Modify Partition Link" HMC operation and
+        waits for completion of its asynchronous job.
+
+        HMC/SE version requirements:
+
+        * SE version >= 2.16.0
+        * for technology-specific support, see the API features described in
+          :ref:`Partition Links`
+
+        Authorization requirements:
+
+        * Object-access permission to the CPC that will be associated with
```

Comment:
You don't need object access permissions for the CPC on modify for an existing partition link. That's only the case when you create a new partition link.

Reply:
DONE